src/niri.rs

```diff
@@ -247,6 +249,7 @@ pub struct OutputState {
     pub lock_render_state: LockRenderState,
     pub lock_surface: Option<LockSurface>,
     pub lock_color_buffer: SolidColorBuffer,
+    pub pending_screencopy: Option<Screencopy>,
```

Comment:
I guess this needs to be a `Vec<Screencopy>` if multiple clients want to copy the same frame.

Reply:
Something like that, yeah. I noticed some differing behaviours in this regard for tty/winit.

To "stress test" the screencopy, i tried running a loop where i call `grim` 12 times. On TTY, this causes the screen to stutter for a while (if i only call it like, say, three times, this is nearly unnoticeable to me - and since this is likely to happen after some "selection of the screen" or such, this is probably okay). However, on winit, it runs fast (???) and i previously observed that it would render the first screenshot fine, the second one sometimes, but the rest were just empty or such. I could not reproduce this behaviour now when i was trying to verify that this really did occur, so either i accidentally fixed it or it was never a real issue. But i'd imagine this kind of thing could be a result of this only storing one `Screencopy`. I didn't implement it as a Vec as i thought this edge case is unlikely, and even if it does occur, you can likely retry.

Reply:
I added this field initially so that it could be accessed during the main render function, but as it stands there is not really any need to do that. Could i just render directly in the screencopy handler now (like i did in the initial implementation)? Is there any reason to prefer doing in the render loop? If we can do it there, then the field can be removed entirely and there is no need for a `Vec`.

Reply:
> Could i just render directly in the screencopy handler now (like i did in the initial implementation)? Is there any reason to prefer doing in the render loop?

I think the intent of screencopy is to get the same frame as will be displayed on screen. For that you need to render right after rendering for screen output. Because if you render in the handler, it might be too early: the mouse may still move until a redraw, or new client buffers may still be committed that will be included in the frame that will be shown on the output.

(To be fair, if the compositor is idling, we can still return frames that are too early. But I can worry about this myself later; this is still the right approach for the final solution.)

Reply:
The previously observed (and magically disappeared) behaviour i described yesterday is probably a side effect of an unrelated bug. In that instance, grim "succeeded" 12 times with sometimes garbage data (this has since gone away). But if i change it to fork each time, then like half of them fail with a non-zero exit code.  Changing `Option` to a `Vec` here allows 12 forked grims to successfully capture the same frame.

An optimization i could imagine, is if there are multiple screencopies with different regions, it might be more efficient to render the whole frame once and only copy different regions. But, that also feels like a case of severe premature optimization. The only realistic use case for many screencopies frequently enough that this matters is screen recording, but then this doesn't apply because they're always on different frames.